File: backend/src/main/java/cohappy/backend/model/dto/GetHouseAdvertesimentDTO.java
package cohappy.backend.model.dto;

import cohappy.backend.model.HouseState;
import lombok.Data;

import java.util.List;

@Data
public class GetHouseAdvertesimentDTO {
    private String houseCode;
    private List<byte[]> images;
    private int costPerMonth;
    private String country;
    private String region;
    private String street;
    private int civicNumber;
    private HouseState state;
    private String publishedByCode;
    private List<byte[]> publishedByImages;
    private String publishedByEmail;
    private String publishedByPhoneNumber;
}

File: frontend/app/src/main/java/cohappy/frontend/model/dto/response/GetHouseAdvertesimentDTO.kt
package cohappy.frontend.model.dto.response

import cohappy.frontend.model.dto.HouseStateEnum
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class GetHouseAdvertesimentDTO(
    var houseCode: String? = null,
    var images: List<ByteArray>? = null,
    var costPerMonth: Int? = null,
    var country: String? = null,
    var region: String? = null,
    var street: String? = null,
    var civicNumber: Int? = null,
    var state: HouseStateEnum? = null,
    var publishedByCode: String? = null,
    var publishedByImages: List<ByteArray>? = null,
    var publishedByEmail: String? = null,
    var publishedByPhoneNumber: String? = null,
    var description: String? = null
)